src/utils/provenance.js

```diff
@@ -0,0 +1,70 @@
+'use strict'
```

Comment:
generally speaking I'm not a fan of programming against specific versions of tools we use, npm in this case, but you did such a good job here at covering the various cases and doing the research that it would be a pity to undo it. so thanks for all this work, very nice